### ISSUE TITLE
Implementation of RFC #1533

### ISSUE THREAD

**Issue by schuyler1d** [ProgressiveCoders/Spoke#1572]
This implementation issue is meant to track different steps on the project proposal of RFC #1533

DONE
* schema changes
* joining ui and campaign invite changes

SKY:
 - [x] panel: Dynamic assignment enabling on campaign (partially done)
  - [ ] max_contacts handling (sky working on this)
- [ ] TexterUI extensions framework
  * admin ui
  * show/notify function
  * component
- [ ] and GraphQL endpoint for texters to receive what is enabled and not
- [ ] Sidebox Mobile UI

HIEMANSHU:
- [ ] Request/Release mutations:
  - [ ] Request is already present in findNewCampaignContact
    * src/server/api/mutations/findNewCampaignContact.js
    * can this be more efficient in assigning?
    * still related to assignment.max_contacts?
  - [ ] Release
    - release needsMessage or all (including messaged) -- different settings will allow one or the other
    - confirm (tests) that user_id is tracked in messages
- [ ] Consequences for TexterTodoList (do any of these buttons need to change -- maybe 'request texts')
- [ ] panel: campaign.features settings and org 'default features'
  - depends on extensions for showing widget UI? (blocked by Sky unless further along with TexterUI extensions)
- [ ] organization Settings panel for default settings (using the same widgets)

FUTURE (start looking at this and think about how to refactor (and make it more testable)):

- [ ] Complicated 'load contacts' logic 
  - containers/AssignmentTexterContact.jsx and 
  - containers/TexterTodo.jsx
  - components/AssignmentTexter/ContactController.jsx
  * Use cases:
    - We can't 'download all the data at once' -- it's too slow
      - needs to be at least partially incremental
      - probably can reduce the calls -- now ~20/50 --> 300
        - especially for needsMessage (first texts)
    - Reassigned more contacts (for needsMessage) while they've already started
      - currently we refresh the contact list and 'overwrite' the array of contact ids
        -- causes some bugs, because so this causes array index issues (see ContactController
  * Goals:
    - More predictable / consistent
    - Fewer API calls if possible
    - (stretch goals -- keeping state for contacts (e.g. current message)
